base/strings/util.jl

```diff
@@ -52,7 +52,8 @@ function endswith(a::AbstractString, b::AbstractString)
 end
 endswith(str::AbstractString, chars::Chars) = !isempty(str) && last(str) in chars
 
-startswith(a::String, b::String) = startswith(a.data, b.data)
+startswith(a::String, b::String) =
+    (a.len >= b.len && ccall(:memcmp, Int32, (Ptr{UInt8}, Ptr{UInt8}, UInt), a, b, b.len) == 0)
 startswith(a::Vector{UInt8}, b::Vector{UInt8}) =
```

Comment:
That brings up a good point: there are several functions here that clearly were only defined on vectors because of the string representation. In fact I removed a couple. We need to decide whether we want to keep them. I favor removing them, since it's a bit random to define some small subset of string functions on byte vectors.